Sol.java

```diff
@@ -0,0 +1,104 @@
+import java.util.*;
+
+public class Sol {
+    int[] in ;
+    HashMap < Integer, Integer > exp;
+    LinkedList < Move > moves_coll;
+    int zeropos;
+
+    public LinkedList<Move> move_cars(int[] in , int[] out) {
```

Comment:
I would suggest you to add some documentation here. It is the only place for which I would say it is a must.